java/src/main/java/com/nedap/retail/example/rest/ApiCaller.java

```diff
@@ -32,45 +35,53 @@
 
     public ApiCaller(final String baseUrl) {
         // make sure baseUrl does not end with a slash
-        if (baseUrl.endsWith("/")) {
-            this.baseUrl = baseUrl.substring(0, baseUrl.length() - 1);
-        } else {
-            this.baseUrl = baseUrl;
+        this.baseUrl = baseUrl;
```

Comment:
Alright noted. I noticed it was checked here, but not in `RenosWebSocketClient` so when a trailing `/` was present in the argument, it would not work. I'll do this for both then.